### PR TITLE
Executable Validator Guide

### PR DESCRIPTION
**Pull request by CarlBeek**
Executifing the validator guide in response to @JustinDrake's #1227.

This does not add any tests, it just adds `0_beacon-chain-validator.md` to `phase0/spec.py` and fixes some typos and bugs in the V-guide to pass linting.

This PR also implements `partial_clean` in the Makefile. This deletes everything except the venvs which is very useful when on limited internet connections.

I will be AFK for the next 5 hours, so I will only be able to look at feedback thereafter.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -29,16 +29,21 @@ COV_INDEX_FILE=$(PY_SPEC_DIR)/$(COV_HTML_OUT)/index.html
 
 all: $(PY_SPEC_ALL_TARGETS) $(YAML_TEST_DIR) $(YAML_TEST_TARGETS)
 
-clean:
+# deletes everything excpet the venvs
```
Comment (protolambda) [inline Makefile]:
```suggestion
# deletes everything except the venvs
```

File: specs/validator/0_beacon-chain-validator.md [hunk 2]
```diff
@@ -217,11 +217,7 @@ def get_epoch_signature(state: BeaconState, block: BeaconBlock, privkey: int) ->
     return bls_sign(
```
Comment (JustinDrake) [inline specs/validator/0_beacon-chain-validator.md]: Now you can shrink `bls_sign` into `bls_sign(privkey, hash_tree_root(slot_to_epoch(block.slot)), domain)`

File: specs/validator/0_beacon-chain-validator.md [hunk 3]
```diff
@@ -205,18 +210,19 @@ Set `block.state_root = hash_tree_root(state)` of the resulting `state` of the `
...
-    message_hash=hash_tree_root(slot_to_epoch(block.slot)),
-    domain=get_domain(
-        fork=fork,  # `fork` is the fork object at the slot `block.slot`
-        epoch=slot_to_epoch(block.slot),
-        domain_type=DOMAIN_RANDAO,
+def get_epoch_signature(state: BeaconState, block: BeaconBlock, privkey: int) -> BLSSignature:
+    return bls_sign(
+        privkey=privkey,  # privkey stored locally
+        message_hash=hash_tree_root(slot_to_epoch(block.slot)),
+        domain=get_domain(
```
[the remark this answers is not in the document]
Comment (CarlBeek) [reply specs/validator/0_beacon-chain-validator.md]: I compressed it to be more in line with what is done in `0_beacon_chain.md`. I don't think breaking the `domain` out into a new variable is warranted here.

[the remark this answers is not in the document]
Comment (JustinDrake) [reply specs/validator/0_beacon-chain-validator.md]: `0_beacon_chain.md` breaks `domain` into a new variable. It's the cleanest IMO.

File: specs/validator/0_beacon-chain-validator.md (hunk 2 above)
Comment (CarlBeek) [reply specs/validator/0_beacon-chain-validator.md]: It's a smaller line count, but I don't think it brings improved clarity.

**Pull request merged**